platform/linux-generic/odp_classification.c

```diff
@@ -1770,6 +1780,32 @@ uint64_t odp_pmr_to_u64(odp_pmr_t hdl)
 	return _odp_pri(hdl);
 }
 
+int odp_cls_queue_stats(odp_cos_t hdl, odp_queue_t queue,
+			odp_cls_queue_stats_t *stats)
+{
+	cos_t *cos = get_cos_entry(hdl);
+	int queue_idx;
+
+	if (odp_unlikely(cos == NULL)) {
+		ODP_ERR("Invalid odp_cos_t handle\n");
+		return -1;
+	}
+
+	queue_idx = _odp_cos_queue_idx(cos, queue);
+	if (odp_unlikely(queue_idx < 0)) {
+		ODP_ERR("Invalid odp_queue_t handle\n");
+		return -1;
+	}
+
+	if (stats)
+		memset(stats, 0, sizeof(odp_cls_queue_stats_t));
+
+	stats->discards = odp_atomic_load_u64(&cos->s.stats[queue_idx].discards);
```

Comment:
May be it better to return failure if stats == NULL.

Now if (stats) check applies only to memset, but not actual stat writes.

Reply:
Fixed in V5.